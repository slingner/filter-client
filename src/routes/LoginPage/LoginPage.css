.LoginPage {
  display: flex;
  flex-direction: column;
  background-image: url(./coffee2.jpg);
  box-shadow: inset 0 0 0 1000px rgba(72, 47, 34, 0.5);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  text-align: center;
  margin: auto;
  padding: 0;
}
