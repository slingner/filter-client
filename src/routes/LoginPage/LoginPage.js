import React, { Component } from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import { Section } from '../../components/Utils/Utils'
import './LoginPage.css'

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  
  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
    this.props.handleLoginSuccess()
  }

  render() {
    return (
      <Section className='LoginPage'>
        <h2 className='login-header'><span aria-label='coffee-emoji' role='img'>☕</span> Login <span aria-label='coffee-emoji' role='img'>☕</span></h2>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </Section>
    )
  }
}
