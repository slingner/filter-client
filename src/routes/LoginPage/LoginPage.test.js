import React from 'react'
import ReactDOM from 'react-dom'
import LoginPage from './LoginPage'
import renderer from 'react-test-renderer';


describe(`<LoginPage />`, () => {
  //Smoke Testing
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <LoginPage 
        
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div)
  })
  //Snapshot Testing
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<LoginPage />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})