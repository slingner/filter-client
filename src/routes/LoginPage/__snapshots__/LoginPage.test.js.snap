// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders UI as expected 1`] = `
<section
  className="Section LoginPage"
>
  <h2
    className="login-header"
  >
    <span
      aria-label="coffee-emoji"
      role="img"
    >
      ☕
    </span>
     Login 
    <span
      aria-label="coffee-emoji"
      role="img"
    >
      ☕
    </span>
  </h2>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="user_name"
    >
      <label
        htmlFor="LoginForm__user_name"
      >
        User name
      </label>
      <input
        className="Input "
        id="LoginForm__user_name"
        name="user_name"
        required={true}
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <input
        className="Input "
        id="LoginForm__password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="Button "
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
